### PR TITLE
[fix] MPI init for unit tests

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by blefaudeux** [comment]
wrong right now, re-reading, this code sure is branchy.. a proper refactor down the line would be great

File: fairscale/utils/testing.py [hunk 1]
```diff
@@ -168,7 +170,7 @@ def spawn_for_all_world_sizes(test_func: Callable, world_sizes: List[int] = get_
 
 
 def worker_process(
-    rank: int, world_size: int, filename: str, filename_rpc: str, func: Callable, args: Any, error_queue: Any
+    rank: int, world_size: int, filename: str, filename_rpc: str, func: Callable, args: Any, error_queue: Any,
```
Comment (anj-s) [inline fairscale/utils/testing.py:173]: Why the extra comma?

Comment (blefaudeux) [reply fairscale/utils/testing.py:173]: autoformat, no idea

Comment (blefaudeux) [reply fairscale/utils/testing.py]: I just removed it, not sure how that happened

**Comment by blefaudeux** [comment]
> wrong right now, re-reading, this code sure is branchy.. a proper refactor down the line would be great

fixed since two commits, testing repeatability :)

**Review by anj-s — APPROVED** [review]

**Pull request merged**